### ISSUE TITLE
Issue with simple external proxing with Harmon

### ISSUE THREAD

**Issue by jamesla** [No9/harmon#3]
Hi I am trying to make a simple external proxy POC however am having issues with Harmon returning strange data.

I have written the following code and simplified it to illustrate what is happening. Ignore the fact that I am passing harmon an empty action as it does the same thing with a valid action (the action does work if it is passed in) however I am omitting it for the sake of simplicity in this example. More details in comments for each server below.

var http = require('http'),
    httpProxy = require('http-proxy');

/\* If a client connects to this server plan text web pages will work fine however most web sites will hang half way through loading, images won't load and will generally display wrong */
httpProxy.createServer(
  require('harmon')([], []), 9000, 'localhost'
).listen(8000);

/\* If client connects directly to this server node http proxy will proxy as expected. - so node http proxy must be working as expected. */
httpProxy.createServer(function (req, res, proxy) {
  proxy.proxyRequest(req, res, {
    host: req.headers.host, 
    port: 80
  });
}).listen(9000);

Basically the page will load however the data will be distorted and sometimes the node server will hang.

I have read through the examples and looked at the source but can't work out what is happening if anybody could enlighten me that would be great!
